### PR TITLE
GODRIVER-303: Provide method to drop collection

### PR DESCRIPTION
**Pull request by igorKryvenko**

### CODE REVIEW

**Comment by skriptble** [comment]
Hi Igor!

Thanks for the contribution. This is a great start and almost complete. Looks like we're just missing the mongo.Collection method `Drop`. This method can directly call the command you made. It only needs to return an error, so the function signature will look like this:

```
func (coll *Collection) Drop(ctx context.Context) error
```

We can just return the error directly, unless the error returned from the `command.DropCollection.RoundTrip`, is a `command.Error` with a `command.Error.Code` of 26 then we should return `nil`.

Let me know if you can make these changes, if not I can add that method and get this all merged in.

Thanks again for your contribution!

--Kris

**Comment by igorKryvenko** [comment]
@skriptble 
Hi. Thanks for the reply. 
Yeah, I can make it.

File: mongo/collection.go [hunk 1]
```diff
@@ -699,3 +699,18 @@ func (coll *Collection) Watch(ctx context.Context, pipeline interface{},
 func (coll *Collection) Indexes() IndexView {
 	return IndexView{coll: coll}
 }
+func (coll *Collection) Drop(ctx context.Context) error {
```
Comment (skriptble) [inline mongo/collection.go]: Can you add a documentation comment here?

File: mongo/collection.go [hunk 2]
```diff
@@ -699,3 +699,18 @@ func (coll *Collection) Watch(ctx context.Context, pipeline interface{},
...
+		ctx = context.Background()
+	}
+
+	cmd := command.DropCollection{
+		DB:         coll.db.name,
+		Collection: coll.name,
+	}
+	_, err := dispatch.DropCollection(ctx, cmd, coll.client.topology, coll.writeSelector)
+	if err != nil {
+		return err
```
Comment (skriptble) [inline mongo/collection.go]: We just need to check if the error is a `command.Error` and if it is if the `Code` property is equal to `26`, if it is, return nil.

**Comment by igorKryvenko** [comment]
@skriptble Done.

**Pull request merged**